buildSrc/src/main/kotlin/spotless-java-extension.gradle.kts

```diff
@@ -24,44 +24,96 @@ plugins {
     }
 }
 
-class MultilineFormatter : Serializable {
-    fun format(content: String): String {
-        val tripleQuote = "\"\"\""
+
+abstract class MultilineFormatter : Serializable {
+    companion object {
+        const val TRIPLE_QUOTES = "\"\"\""
+        fun createStep(name : String, formatter : MultilineFormatter) : FormatterStep {
+            return FormatterStep.create(
+                name,
+                formatter
+            ) { FormatterFunc { content -> it.format(content) } }
+        }
+    }
+
+    abstract fun format(content : String) : String
+}
+
+class MultilineIndentFormatter : MultilineFormatter() {
+    override fun format(content : String) : String {
         val lines = content.lines()
         val result = StringBuilder()
         var i = 0
         while (i < lines.size) {
             val line = lines[i]
-            if (!line.trimEnd().endsWith(tripleQuote)) {
+            if (!line.trimEnd().endsWith(TRIPLE_QUOTES)) {
                 result.append(line)
-                if (i + 1 < lines.size) result.append("\n")
+                if (i + 1 < lines.size) result.append('\n')
                 i++
                 continue
             }
-            val baseIndent = line.indexOf(tripleQuote)
-            result.append(line).append("\n")
+            val baseIndent = line.indexOf(TRIPLE_QUOTES)
+            result.append(line).append('\n')
             i++
             val multilineStringLines = mutableListOf<String>()
             while (i < lines.size) {
                 val multilineStringLine = lines[i++]
                 multilineStringLines.add(multilineStringLine)
-                if (multilineStringLine.contains(tripleQuote)) break
+                if (multilineStringLine.contains(TRIPLE_QUOTES)) break
             }
             val minIndent =
                 multilineStringLines
                     .filter { it.isNotBlank() }
-                    .map { l -> l.indexOfFirst { ch -> !ch.isWhitespace() }.takeIf { it >= 0 } ?: line.length }
-                    .minOrNull() ?: 0
-            multilineStringLines.forEach { blockLine ->
-                result.append(" ".repeat(baseIndent)).append(blockLine.drop(minIndent)).append("\n")
+                    .minOfOrNull { l -> l.indexOfFirst { ch -> !ch.isWhitespace() }.takeIf { it >= 0 } ?: line.length }
+                    ?: 0
+            multilineStringLines.forEach { result.append(" ".repeat(baseIndent)).append(it.drop(minIndent)).append('\n') }
+        }
+        return result.toString()
+    }
+}
+
+class MultilineWithParametersFormatter : MultilineFormatter() {
+    override fun format(content : String) : String {
+        val lines = content.lines()
+        val result = StringBuilder()
+        var i = 0
+        while (i < lines.size) {
+            val line = lines[i]
+            if (line.trimEnd().endsWith(TRIPLE_QUOTES) && i + 1 < lines.size
+                && lines[i + 1].trimStart().startsWith(".formatted(")) {
```

Comment:
This code seems too specific - a format should apply to all methods, not just this `formatted` method.